src/main/java/com/redhat/devtools/lsp4ij/BooleanRef.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Reference to a boolean value that simplifies mutation of that value in lambdas/inner classes.
+ */
+@ApiStatus.Internal
+public class BooleanRef extends MutableRef<Boolean> {
```

Comment:
Several mutable reference types to provide indirect access to a mutable value of the respective simple types in lambdas. For `boolean` values we want to be able to perform logical ANDs and ORs; for `string` values we want to be able to append; for enumerated constant values we want to be able to retain the highest/lowest-ordered value.